File: java/parser/src/main/java/gov/lbl/parser/domain/Primary.java
package gov.lbl.parser.domain;

import java.util.Collection;

public class Primary {
    private Double unsigned_number;
    private String string;
    private Boolean is_false;
    private Boolean is_true;
    private Function_call_primary function_call_primary;
    private Component_reference component_reference;
    private Output_expression_list output_expression_list;
    private Collection<Expression_list> expression_lists;
    private Function_arguments function_arguments;

    public Primary(Double unsigned_number, String string, Boolean is_false, Boolean is_true, Function_call_primary function_call_primary, Component_reference component_reference,
                    Output_expression_list output_expression_list, Collection<Expression_list> expression_lists, Function_arguments function_arguments){
        this.unsigned_number = unsigned_number;
        this.string = string;
        this.is_false = is_false;
        this.is_true = is_true;
        this.function_call_primary = function_call_primary;
        this.component_reference = component_reference;
        this.output_expression_list = output_expression_list;
        this.expression_lists = expression_lists;
        this.function_arguments = function_arguments;
    }
}

File: jsParser/domain/Primary.js
class Primary {
  constructor (unsigned_number, primary_string, is_false, is_true, function_call_primary, component_reference, output_expression_list, expression_lists, function_arguments, end) {
    unsigned_number != null ? this.unsigned_number = unsigned_number : ''
    primary_string != null && primary_string != '' ? this.primary_string = primary_string : ''
    is_false != null ? this.is_false = is_false : ''
    is_true != null ? this.is_true = is_true : ''
    function_call_primary != null ? this.function_call_primary = function_call_primary : ''
    component_reference != null ? this.component_reference = component_reference : ''
    output_expression_list != null ? this.output_expression_list = output_expression_list : ''
    expression_lists != null ? expression_lists.length != 0 ? this.expression_lists = expression_lists : '' : ''
    function_arguments != null ? this.function_arguments = function_arguments : ''
    end != null ? this.end = end : ''
  }
}
exports.Primary = Primary
